pysat/tests/test_utils_testing.py

```diff
@@ -82,3 +95,112 @@ def test_nan_equal_bad(self, val1, val2):
+
+        Parameters
+        ----------
+        warn_type : Warning
+            Warning class to be raised
+
+        """
+        warn_msg = 'test warning'
+
+        # Raise the desired warning
+        with warnings.catch_warnings(record=True) as war:
+            warnings.warn(warn_msg, warn_type)
+
+        # Evaluate the warning output
+        testing.eval_warnings(war, [warn_msg], warn_type)
+        return
+
+    @pytest.mark.parametrize("warn_type", [
+        UserWarning, DeprecationWarning, SyntaxWarning, RuntimeWarning,
+        FutureWarning, PendingDeprecationWarning, ImportWarning,
+        UnicodeWarning, BytesWarning, ResourceWarning])
+    def test_eval_warnings_bad_type(self, warn_type):
+        """Test warning evaluation function failure for mismatched type.
+
+        Parameters
+        ----------
+        warn_type : Warning
+            Warning class to be raised
+
+        """
+        warn_msg = 'test warning'
+        bad_type = UserWarning if warn_type != UserWarning else BytesWarning
+
+        # Raise the desired warning
+        with warnings.catch_warnings(record=True) as war:
+            warnings.warn(warn_msg, warn_type)
+
+        # Catch and evaluate the expected error
+        with pytest.raises(AssertionError) as aerr:
+            testing.eval_warnings(war, [warn_msg], bad_type)
+
+        assert str(aerr).find('bad warning type for message') >= 0
+        return
+
+    @pytest.mark.parametrize("warn_type", [
+        UserWarning, DeprecationWarning, SyntaxWarning, RuntimeWarning,
+        FutureWarning, PendingDeprecationWarning, ImportWarning,
+        UnicodeWarning, BytesWarning, ResourceWarning])
+    def test_eval_warnings_bad_msg(self, warn_type):
+        """Test warning evaluation function failure for mismatched message.
+
+        Parameters
+        ----------
+        warn_type : Warning
+            Warning class to be raised
+
+        """
+        warn_msg = 'test warning'
+        bad_msg = 'not correct'
+
+        # Raise the desired warning
+        with warnings.catch_warnings(record=True) as war:
+            warnings.warn(warn_msg, warn_type)
+
+        # Catch and evaluate the expected error
+        with pytest.raises(AssertionError) as aerr:
+            testing.eval_warnings(war, [bad_msg], warn_type)
+
+        assert str(aerr).find('did not find') >= 0
+        return
+
+    def test_prep_dir_exists(self):
+        """Test successful pass at creating existing directory."""
+
+        # Create a temporary directory
+        tempdir = tempfile.TemporaryDirectory()
+        assert os.path.isdir(tempdir.name)
+
+        # Assert prep_dir does not re-create the directory
+        assert not testing.prep_dir(tempdir.name)
+
+        # Clean up temporary directory
+        tempdir.cleanup()
+        return
+
+    def test_prep_dir_new(self):
+        """Test successful pass at creating existing directory."""
+
+        # Create a temporary directory and get it's name
```

Comment:
```suggestion
        # Create a temporary directory and get its name
```
https://www.faa.gov/about/initiatives/plain_language/articles/its/